contracts/contracts.go

```diff
@@ -42,7 +42,7 @@ func (p *GatewayProvider) declareAndWaitWithWallet(ctx context.Context, compiled
 		return nil, err
 	}
 	if !receipt.Status.IsTransactionFinal() ||
```

Comment:
I noticed that `IsTransactionFinal` still includes the `REJECTED` state (even after failed tx inclusion). Is it because
a. this state is temporarily reported by the gateway for all txs for backward compatibility 
or 
b. will the gateway have to support `REJECTED` forever for old transactions?

Reply:
The rpc spec (v4) no longer has TXN_STATUS, so this function should be removed. It's probably worth highlighting that the contracts package is outdated, and likely needs updated to follow the spec. Also, any functions / packages to do with the gateway will be removed because the sequencer will be going offline.

Reply:
Thanks for clarifying, I originally thought the old rpc (< v4) was there for backwards compatibility but good to know it can be removed entirely!